core/utils/wasm-builder/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # WASM builder is an utility for building a project as a WASM binary
+//!
+//! The WASM builder is a tool that integrates the building of a WASM binary of your project into the main
+//! `cargo` build process.
+//!
+//! ## Project setup
+//!
+//! A project that should be compiled as WASM binary needs to add a `build.rs`, add
```

Comment:
I would suggest we do a numbered list here:

```
//! A project that should be compiled as WASM binary needs to:
//! 1. Add a `build.rs` file
//! 2. Add `substrate-wasm-builder-runner` as dependency into `build-dependencies`
//! 3. Add a feature called `no-std`
//!
//! The `build.rs` needs to contain the following code:

Reply:
```
//! A project that should be compiled as a WASM binary needs to:
//!
//! 1. Add a `build.rs` file.
//! 2. Add `substrate-wasm-builder-runner` as dependency into `build-dependencies`.
//! 3. Add a feature called `no-std`.
//!
//! The `build.rs` file needs to contain the following code:
```